### ISSUE TITLE
Domains created/deleted through the CLI are not listed/unlisted in the webadmin

### ISSUE THREAD

**Issue by tituspijean** [YunoHost/issues#2068]
### Describe the bug

Domains created or deleted through the CLI are not appearing in or disappearing from the webadmin unless:
- another domain is altered through the webadmin
- the YunoHost API service is restarted

### Context

- Hardware: any?
```
yunohost -v
yunohost:
  repo: stable
  version: 11.0.9.12
yunohost-admin:
  repo: stable
  version: 11.0.10
moulinette:
  repo: stable
  version: 11.0.9
ssowat:
  repo: stable
  version: 11.0.9
```
- I have access to my server: SSH and webadmin
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: no

### To reproduce

1. Create domain through the CLI: `yunohost domain add test.tld`
2. Refresh webadmin: the domain is not listed
3. `yunohot domain list`: the domain is listed
4. Either restart the API service or create a domain with the webadmin
5. The test domain appears

### Expected behavior

Domain API list should be updated when using CLI

### Logs

`yunohost domain add... ` : https://paste.yunohost.org/raw/vamizoyaho (sorry, plenty of domains in that server...)

**Comment by alexAubin** [comment]
Hmyeah that's because of caching : https://github.com/YunoHost/yunohost/blob/dev/src/domain.py#L51

the `yunohost-api` process has its own cache, but no cache invalidation is triggered when adding a domain from the CLI because that's another process ...

One way to mitigate this issue would be to set eg. an N-sec (for example 15s) cache expiration mechanism

**Comment by mose** [comment]
We noticed in our tests hunting this bug that restarting the API was solving the problem, so I guess it rebuilds cache at startup. Maybe an API restart could be added on the CLI command that changes domains ?

**Comment by alexAubin** [comment]
Eeeeh definitely not, because sometimes such operation could be directly or indirectly triggered from the API itself so the API would kill itself in the middle of a job ...

**Comment by alexAubin** [comment]
Should be ~fixed in 11.1 because we added an expiration for the domain cache : https://github.com/YunoHost/yunohost/blob/dev/src/domain.py#L51
